README.md

```diff
@@ -177,26 +180,35 @@ But even with 16 bit pointers (65,536) are probably just enough for most NLP tas
 
 One other thing to do is to use [MLLabelUtils.jl](https://github.com/JuliaML/MLLabelUtils.jl) (on-top of `InternedStrings.jl`) and encode your strings as `Int`s.
 
-### What about symbols? Aren't they interned strings?
+### What about Symbols? Aren't they interned strings?
 
 Symbols are not semantically strings at all.
 Semantically they are names.
 They are not a subtype of `AbstractString`, they don't support string operations.
 You could build a interned string around them though.
 However, to the best of my knowledge, they are never garbage collected.
 
-### What about the Factor packages
+### What about serialization ?
+
+The interning is transparent: after interning it is still just a string.
+So it serializes completely fine -- as the string that it is.
+However, when you deserialize a string it creates a new (not yet interned) string with the serialized content.
+This means interning strings in other processes (e.g. via `pmap`) does not work.
+(It is however completely safe to intern strings in any thread. This code is fully thread-safe.)
+
+
+### What about the factor packages ?
 
-`CategorialArray`s etc  are pretty similar.
+`CategorialArray`s etc.  are pretty similar.
 But they are focused on pooling for a single array.
 
 The unmaintained (and unregistered) [PooledElements.jl](https://github.com/tshort/PooledElements.jl), did global pools but without automatic garbage collection.
 Also not referentially sane -- magic is required to make sure it was working with serialization etc.
 
 ### What is the downside?
 
-There is basically no downside to interning a String.
-The only overhead is to compute the hash of the string if it has already been interned or not.
+There is basically no downside to interning a string.
+The only overhead is to compute the hash of the string, to check if it has already been interned or not.
```

Comment:
That brings up an important issue, Strs will suffer because `hash` is forced to calculate a `String` compatible hash.
However, you *could* use a hash function other than hash (and fall back to `hash`) so that other string types could use a faster hashing algorithm.   Something to think about, I'm not sure what the best approach would be.